students/sumin-hwang/westagram/users/views.py

```diff
@@ -0,0 +1,123 @@
+import json
+import re
+
+from django.http import JsonResponse
+from django.views import View
+from django.db.models import Q
+
+from users.models import User
+
+PASSWORD_MINIMUM_LENGTH = 8
+
+class SignUpView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        mobile_number = data.get('mobile_number', None)
+        email         = data.get('email', None)
+        name          = data.get('name', None)
+        user_name     = data.get('user_name', None)
+        password      = data.get('password', None)
+        
+        mobile_number_form = re.compile('[0-9]{3}-[0-9]{3,4}-[0-9]{4}')
+        email_form         = re.compile('^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+
+        if not (
+                (mobile_number or email)
+                and name
+                and user_name
+                and password
+                ):
+            return JsonResponse({'MESSAGE':'INVALID_KEY'}, status=400)
+        
+        elif mobile_number:
+            print(mobile_number_form.match(mobile_number))
+            if not mobile_number_form.match(mobile_number):
+               return JsonResponse({'MESSAGE':'INVALID_MOBILE_NUMBER'}, status=400)
+            elif User.objects.filter(mobile_number = data['mobile_number']):
+               return JsonResponse({'MESSAGE':'ALREADY_EXISTS'}, status=400)
+
+        elif email:
+           if not email_form.match(str(email)):
+               return JsonResponse({'MESSAGE':'INVALID_EMAIL'}, status=400)
+           elif User.objects.filter(email = data['email']):
+               return JsonResponse({'MESSAGE':'ALREADY_EXISTS'}, status=400)
+        
+        # if mobile_number:
+        #     if '-' in data['mobile_number']:
+        #         return JsonResponse({'MESSAGE':'INVALID_MOBILE_NUMBER'}, status=400)
+        #     if User.objects.filter(mobile_number = data['mobile_number']):
+        #         return JsonResponse({'MESSAGE':'ALREADY_EXISTS'}, status=400)
+        
+        # if email:
+        #     if ('@' or '.') not in data['email']:
+        #         return JsonResponse({'MESSAGE':'INVALID_EMAIL'}, status=400)
+        #     if User.objects.filter(email = data['email']):
+        #         return JsonResponse({'MESSAGE':'ALREADY_EXISTS'}, status=400)
+
+        
+        if User.objects.filter(user_name = data['user_name']):
```

Comment:
```suggestion
        if User.objects.filter(user_name = data['user_name']).exists():
```
filter을 사용해서 존재 여부 판단 잘 하셨습니다.
보통 존재 여부만 판단 할 때는 .exists()를 붙여서 사용합니다.
filter을 사용할 경우 조건에 해당되는 데이터를 가져오지만, .exists()을 붙이면 존재 여부만 판단하고 데이터는 가져오지 않기 때문에 더 효율적입니다.